src/term/mod.rs

```diff
@@ -42,6 +44,9 @@ pub struct RenderableCellsIter<'a> {
     line: Line,
     column: Column,
     selection: Option<RangeInclusive<index::Linear>>,
+    custom_cursor_colors: bool,
```

Comment:
Instead of having all of these properties on the iterator, I think things can be dramatically simplified by having an `cursor_original: Option<IndexedCell>` field instead. In the iterator's initialize method, this can be set to the *original* cell at the cursor location.

```rust
    self.cursor_original = Some(IndexedCell {
        line: cursor.line,
        col: cursor.col,
        cell: grid[cursor]
    });
```

In the drop method, the cell at `grid[cursor]` would be replaced by this value. An option is used here since this wouldn't be necessary if the cursor isn't visible.